src/dbally/prompt/aggregation.py

```diff
@@ -0,0 +1,122 @@
+from typing import List, Optional
+
+from dbally.audit import EventTracker
+from dbally.exceptions import UnsupportedAggregationError
+from dbally.iql import IQLQuery
+from dbally.llms.base import LLM
+from dbally.llms.clients import LLMOptions
+from dbally.prompt.template import PromptFormat, PromptTemplate
+from dbally.views.exposed_functions import ExposedFunction
+
+
+def _validate_agg_response(llm_response: str) -> str:
+    """
+    Validates LLM response to IQL
+
+    Args:
+        llm_response: LLM response
+
+    Returns:
+        A string containing aggregations.
+
+    Raises:
+        UnsupportedAggregationError: When IQL generator is unable to construct a query
+        with given aggregation.
+    """
+    if "unsupported query" in llm_response.lower():
+        raise UnsupportedAggregationError
+    return llm_response
+
+
+class AggregationPromptFormat(PromptFormat):
+    """
+    Aggregation prompt format, providing a question and aggregation to be used in the conversation.
+    """
+
+    def __init__(
+        self,
+        question: str,
+        aggregations: List[ExposedFunction] = None,
+    ) -> None:
+        super().__init__()
+        self.question = question
+        self.aggregations = "\n".join([str(aggregation) for aggregation in aggregations]) if aggregations else []
+
+
+class AggregationFormatter:
```

Comment:
This class is generator to me. I'd put it in `iql_generator` module.

Reply:
These changes were suggested by @ludwiktrammer and as the aggregations are not IQL-generated directly I decided to keep it in a separate, dedicated module.

Reply:
Aggregations are part of the IQL syntax, and the fact that we generate them separately doesn't exclude them from the IQL language. This class reflects the behavior of `IQLGenerator`, so copying this logic to the `prompt` module is not correct for me. Perhaps we should have 2 IQL generators - `FilterGenerator` and `AggregationGenerator` to make it cleaner.

Reply:
@mhordynski - what is your perspective from architectural point of view?

Reply:
I agree with having `IQLGenerator` and separating later to `IQLFiltersGenerator` / `IQLAggegationGenerator`. from my perspective aggregations, along other features like `Context()` are becoming parts of the IQL